assets/js/Ioda/components/header/Header.js

```diff
@@ -88,47 +88,40 @@ const languageOptions = [
   { value: "fa", label: "Farsi" },
 ];
 
-class Header extends Component {
-  constructor(props) {
-    super(props);
+const Header = () => {
 
-    this.state = {
-      language: getSavedLanguagePreference(),
-      advancedMode: getSavedAdvancedModePreference(),
-      showDrawer: false,
-      mounted: false,
-    };
-  }
+  const [language, setLanguage] = useState(getSavedLanguagePreference());
```

Comment:
If we're using React v18+, we may be able to use `useSyncExternalStore`, which provides a more robust method for reading data from external systems (like `localStorage`). I wrote a blog post on this: https://www.arvinpoddar.com/blog/syncing-local-storage-with-react-state

Reply:
Thank you for the suggestion. I'll be sure to read your blog post for a more in-depth understanding.